components/style/properties/helpers/animated_properties.mako.rs

```diff
@@ -2521,6 +2529,110 @@ impl ToAnimatedZero for IntermediateSVGPaintKind {
     }
 }
 
+impl Into<NumberOrPercentage> for NonNegativeLengthOrPercentage {
+    fn into(self) -> NumberOrPercentage {
+        self.0.into()
+    }
+}
+
+impl Into<NumberOrPercentage> for NonNegativeNumber {
+    fn into(self) -> NumberOrPercentage {
+        self.0.into()
+    }
+}
+
+impl Into<NumberOrPercentage> for LengthOrPercentage {
+    fn into(self) -> NumberOrPercentage {
+        match self {
+            LengthOrPercentage::Length(len) => NumberOrPercentage::Number(len.to_f32_px()),
+            LengthOrPercentage::Percentage(p) => NumberOrPercentage::Percentage(p),
+            LengthOrPercentage::Calc(_) => {
+                // TODO: We need to treat calc value.
+                // https://bugzilla.mozilla.org/show_bug.cgi?id=1386967
+                debug_assert!(false, "We can't convert to NumberOrPercentage from LengthOrPercentage");
```

Comment:
That should be a panic, and actually this should not even be in a `From` or `Into` impl given these should never fail.

Could you make an intrinsic method instead of implementing this trait, please?